.github/workflows/auto_branching.yml

```diff
@@ -0,0 +1,294 @@
+        with:
+          branch: ${{ github.event.inputs.target_branch }}
+
+      - name: Create label for the ${{ github.event.inputs.target_branch }} branch
+        id: create-label
+        run: |
+          curl -X POST \
+            -H "Authorization: token ${{ secrets.CHERRYPICK_PAT }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/${{ github.repository }}/labels \
+            -d "{\"name\":\"${{ github.event.inputs.target_branch }}\",\"color\":\"fbca04\"}"
+
+      - name: Switch to ${{ github.event.inputs.target_branch }} branch
+        run: git checkout -b "${{ github.event.inputs.target_branch }}"
+
+      - name: Checkout from ${{ github.event.inputs.target_branch }} branch for auto-branching changes
+        id: checkout-to-auto-branch
+        run: |
+            branch_name="auto-branching-${{ github.event.inputs.target_branch }}-$(date '+%s')"
+            git checkout -b "$branch_name"
+            echo "branch_name=$branch_name" >> $GITHUB_OUTPUT
+
+      - name: Update target branch label in dependabot yml file
+        id: update-dependabot
+        run: |
+            # Read the dependabot.yml file
+            FILE_PATH="./.github/dependabot.yml"
+            TARGET_BRANCH="${{ github.event.inputs.target_branch }}"
+            # Append the target branch label to the labels node
+            awk -v target="'$TARGET_BRANCH'" '/^ *labels:/ {$0 = $0 "\n      - " target} 1' "$FILE_PATH" > temp.yml && mv temp.yml "$FILE_PATH"
+
+      - name: Update repository URLs in requirements.txt
+        id: update-repo-urls
+        run: |
+            # Define the file path
+            FILE_PATH="./requirements.txt"
+            # Define the replacement strings
+            replacements=(
+              "airgun @ git+https://github.com/SatelliteQE/airgun.git@master#egg=airgun|airgun @ git+https://github.com/SatelliteQE/airgun.git@${{ github.event.inputs.target_branch }}#egg=airgun"
+              "nailgun @ git+https://github.com/SatelliteQE/nailgun.git@master#egg=nailgun|nailgun @ git+https://github.com/SatelliteQE/nailgun.git@${{ github.event.inputs.target_branch }}#egg=nailgun"
+            )
+            # Create a temporary file
+            TEMP_FILE=$(mktemp)
+            # Perform replacements using a for loop
+            for replacement in "${replacements[@]}"; do
+              old_url=$(echo "$replacement" | cut -d'|' -f1)
+              new_url=$(echo "$replacement" | cut -d'|' -f2)
+              sed "s|${old_url}|${new_url}|g" "$FILE_PATH" > "$TEMP_FILE" && mv "$TEMP_FILE" "$FILE_PATH"
+            done
+
+      - name: Remove the dispatch release GHA
+        id: remove-dispatch-release-gha
+        run: |
+            rm -rf ./.github/workflows/dispatch_release.yml
+            rm -rf ./.github/workflows/auto_branching.yml
+
+      - name: Remove lines with @pytest.mark.stream
+        id: remove-mark-stream
+        run: |
```

Comment:
Can we also add a step to remove codeowner config from zstream branches where we won't need t2 and t1-reviews? It could stop spamming about duplicate notifications for cherrypick PRs in reviewers' inboxes :)

Reply:
It's unrelated to this PR, we can create separate PR to remove from the z-stream cherry-picking branches.

Reply:
We can create a separate PR to remove it from 6.15.z, but I'm proposing if we could auto-remove this while doing auto-branching with this GHA 
CC @jyejare @JacobCallahan

Reply:
Gaurav has a point here. This is something that we discussed in the SatQE Reviewers call.
If we agree that codeowners file is not needed in .z branches, I would like this step to be included in this PR.
@jyejare Can you share your point of view on it here, please?

Reply:
@ogajduse @Gauravtalreja1 do you mean a file should be excluded while auto-branching? While I was reading I felt like that it currently exits need to remove. If that file going to added part to auto-branching I can add that step

Reply:
updated!

Reply:
Good discussion. While I agree codeowners likely to spam reviewers for cherrypicks but those cherrypicks arent being automerged. We need an attention from reviewers to even review the cherrypicks.

Personally I would like to keep it.

Reply:
Agree! We will discuss this in detail in upcoming meetings and decide the best strategy that applies and considered by everyone.

Reply:
I have removed based on @jyejare and @ogajduse comment